File: crates/im-core-dart/src/dto/secure.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum DartDirectSecureState {
    Ready,
    Preparing,
    WaitingForPeer,
    NeedsRepair,
    Unavailable,
    Unknown,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DartDirectSecureStatus {
    pub peer: String,
    pub resolved_peer: Option<String>,
    pub state: DartDirectSecureState,
    pub can_send_secure: bool,
    pub pending_outbox_count: u32,
    pub problem: Option<DartSecureProblem>,
    pub warnings: Vec<String>,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum DartGroupSecureState {
    Ready,
    Syncing,
    NeedsRepair,
    WaitingForMembershipUpdate,
    MissingLocalState,
    Unavailable,
    Unknown,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DartGroupSecureStatus {
    pub group: String,
    pub state: DartGroupSecureState,
    pub can_send_secure: bool,
    pub local_readiness: DartGroupSecureLocalReadiness,
    pub pending_work: DartGroupSecurePendingWork,
    pub problem: Option<DartSecureProblem>,
    pub warnings: Vec<String>,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DartGroupSecureLocalReadiness {
    pub has_local_state: bool,
    pub has_active_membership: bool,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DartGroupSecurePendingWork {
    pub pending_notices: u32,
    pub pending_commits: u32,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DartGroupSecurePrepareResult {
    pub group: String,
    pub state: DartGroupSecureState,
    pub can_send_secure: bool,
    pub warnings: Vec<String>,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DartGroupSecureRepairResult {
    pub group: String,
    pub state: DartGroupSecureState,
    pub repaired: bool,
    pub added_devices: u32,
    pub removed_devices: u32,
    pub remaining_devices: u32,
    pub problem: Option<DartSecureProblem>,
    pub warnings: Vec<String>,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DartSecureProblem {
    pub code: DartSecureProblemCode,
    pub message: String,
    pub retryable: bool,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum DartSecureProblemCode {
    IdentityNotReady,
    PeerNotFound,
    PeerKeysUnavailable,
    SessionNeedsRepair,
    GroupStateUnavailable,
    LocalStateUnavailable,
    TransportUnavailable,
    Unsupported,
    Unknown,
}

File: packages/awiki_im_core/lib/src/models/secure.dart
enum DirectSecureState {
  ready,
  preparing,
  waitingForPeer,
  needsRepair,
  unavailable,
  unknown,
}

class DirectSecureStatus {
  const DirectSecureStatus({
    required this.peer,
    this.resolvedPeer,
    required this.state,
    required this.canSendSecure,
    required this.pendingOutboxCount,
    this.problem,
    this.warnings = const [],
  });

  final String peer;
  final String? resolvedPeer;
  final DirectSecureState state;
  final bool canSendSecure;
  final int pendingOutboxCount;
  final SecureProblem? problem;
  final List<String> warnings;
}

enum GroupSecureState {
  ready,
  syncing,
  needsRepair,
  waitingForMembershipUpdate,
  missingLocalState,
  unavailable,
  unknown,
}

class GroupSecureStatus {
  const GroupSecureStatus({
    required this.group,
    required this.state,
    required this.canSendSecure,
    required this.localReadiness,
    required this.pendingWork,
    this.problem,
    this.warnings = const [],
  });

  final String group;
  final GroupSecureState state;
  final bool canSendSecure;
  final GroupSecureLocalReadiness localReadiness;
  final GroupSecurePendingWork pendingWork;
  final SecureProblem? problem;
  final List<String> warnings;
}

class GroupSecureLocalReadiness {
  const GroupSecureLocalReadiness({
    required this.hasLocalState,
    required this.hasActiveMembership,
  });

  final bool hasLocalState;
  final bool hasActiveMembership;
}

class GroupSecurePendingWork {
  const GroupSecurePendingWork({
    required this.pendingNotices,
    required this.pendingCommits,
  });

  final int pendingNotices;
  final int pendingCommits;
}

class GroupSecurePrepareResult {
  const GroupSecurePrepareResult({
    required this.group,
    required this.state,
    required this.canSendSecure,
    this.warnings = const [],
  });

  final String group;
  final GroupSecureState state;
  final bool canSendSecure;
  final List<String> warnings;
}

class GroupSecureRepairResult {
  const GroupSecureRepairResult({
    required this.group,
    required this.state,
    required this.repaired,
    this.addedDevices = 0,
    this.removedDevices = 0,
    this.remainingDevices = 0,
    this.problem,
    this.warnings = const [],
  });

  final String group;
  final GroupSecureState state;
  final bool repaired;
  final int addedDevices;
  final int removedDevices;
  final int remainingDevices;
  final SecureProblem? problem;
  final List<String> warnings;
}

class SecureProblem {
  const SecureProblem({
    required this.code,
    required this.message,
    required this.retryable,
  });

  final SecureProblemCode code;
  final String message;
  final bool retryable;
}

enum SecureProblemCode {
  identityNotReady,
  peerNotFound,
  peerKeysUnavailable,
  sessionNeedsRepair,
  groupStateUnavailable,
  localStateUnavailable,
  transportUnavailable,
  unsupported,
  unknown,
}
